src/west/manifest.py

```diff
@@ -1288,6 +1292,13 @@ def as_frozen_yaml(self, **kwargs) -> str:
     def projects(self) -> List[Project]:
         return self._projects
 
+    @property
+    def filename(self):
```

Comment:
Since this is API, it needs a docstring, even if it's just a short one.

Reply:
I actually thought about removing this since it's currently unused. I added it initially before I realized that every potential user has that information in a variable already anyway.

Are there tools/plugins which might benefit from such APIs or is it all internal?

Reply:
> Are there tools/plugins which might benefit from such APIs or is it all internal?

The west APIs are separately documented and maintained, yes: https://docs.zephyrproject.org/latest/guides/west/west-apis.html#west-apis

There are various users both upstream (e.g. zephyr/scripts/zephyr_module.py) and in downstream extension commands.

> I actually thought about removing this since it's currently unused.

Go for removing it, we can always add things when we need them later.